jira/client.py

```diff
@@ -2483,7 +2483,10 @@ def _get_url(self, path, base=JIRA_BASE_URL):
 
     def _get_json(self, path, params=None, base=JIRA_BASE_URL):
         url = self._get_url(path, base)
-        r = self._session.get(url, params=params)
```

Comment:
I don't think we should be conditionally switching from get to post here like this.

Reply:
@264768502 : Is there any reason not to just POST all the time?

Reply:
I get jql from filter first, them try to use jql to do issue search
JIRA server feedback input parameter with invalid string.
After switch to use post, search success.
But it has a little strange, after I switch another os, issue go. I have no idea why, maybe use different requests.

My commit maybe not best resolution.
As you know,  http url has length limitation
and 'get' need encode  parameter to url, this will involve some unknown issue.

how about add parameter to let user control use post or get?
In my opinion, check url will give a little hard to maintain

@asqui I don't want change original design too much
Only switch to post when search issue.